src/nxt_http_return.c

```diff
@@ -177,15 +174,16 @@ nxt_http_return_send_ready(nxt_task_t *task, void *obj, void *data)
     action = ctx->action;
     conf = action->u.conf;
 
-    if (!conf->loc_is_const) {
-        ret = nxt_http_return_encode(r->mem_pool, &ctx->encoded,
-                                     &ctx->location);
-        if (nxt_slow_path(ret == NXT_ERROR)) {
-            goto fail;
+    if (conf->location != NULL) {
+
+        if (!nxt_var_is_const(conf->location)) {
```

Comment:
Since If it's const, the `ctx.location` is empty.
encode only ctx.location is not empty.
```
if (ctx->location.length > 0) {
            ret = nxt_http_return_encode(r->mem_pool, &ctx->encoded,
                                         &ctx->location);
            if (nxt_slow_path(ret == NXT_ERROR)) {
                goto fail;
            }
        }
```

Reply:
Ahh, great catch!

That implicitly optimizes for the following case at no cost:

```json
"action": {
    "return": 301,
    "location": "$arg_foo"
}
```